1

```diff
@@ -0,0 +1,204 @@
+        }
+        virtual ~JSONObjectFactory() {
+        }
+
+    public:
+        Core::ProxyType<Core::JSON::IElement> Element(const string&) {
+            return (Core::proxy_cast<Core::JSON::IElement>(Core::ProxyPoolType<StreamJsonTest::Command>::Element()));
+        }
+    };
+
+    template<typename INTERFACE>
+    class JSONConnector : public Core::StreamJSONType<Core::SocketStream, JSONObjectFactory&, INTERFACE> {
+    private:
+        typedef Core::StreamJSONType<Core::SocketStream, JSONObjectFactory&, INTERFACE> BaseClass;
+
+        JSONConnector();
+        JSONConnector(const JSONConnector& copy);
+        JSONConnector& operator=(const JSONConnector&);
+
+    public:
+        JSONConnector(const Core::NodeId& remoteNode)
+            : BaseClass(5, _objectFactory, false, remoteNode.AnyInterface(), remoteNode, 1024, 1024)
+            , _serverSocket(false)
+            , _dataPending(false, false)
+            , _objectFactory(1)
+        {
+        }
+        JSONConnector(const SOCKET& connector, const Core::NodeId& remoteId, Core::SocketServerType<JSONConnector<INTERFACE>>*)
+            : BaseClass(5, _objectFactory, false, connector, remoteId, 1024, 1024)
+            , _serverSocket(true)
+            , _dataPending(false, false)
+            , _objectFactory(1)
+        {
+        }
+        virtual ~JSONConnector()
+        {
+        }
+
+    public:
+        virtual void Received(Core::ProxyType<Core::JSON::IElement>& newElement)
+        {
+            string textElement;
+            newElement->ToString(textElement);
+
+            if (_serverSocket)
+                this->Submit(newElement);
+            else {
+                _dataReceived = textElement;
+                _dataPending.Unlock();
+            }
+        }
+        virtual void Send(Core::ProxyType<Core::JSON::IElement>& newElement)
+        {
+        }
+        virtual void StateChange()
+        {
+            if (this->IsOpen()) {
+                if (_serverSocket)
+                    StreamJsonTest::g_done = true;
+            }
+        }
+        virtual bool IsIdle() const
+        {
+            return (true);
+        }
+        int Wait() const
+        {
+            return _dataPending.Lock();
+        }
+        void Retrieve(string& text)
+        {
+            text = _dataReceived;
+            _dataReceived.clear();
+        }
+    private:
+        bool _serverSocket;
+        string _dataReceived;
+        mutable Core::Event _dataPending;
+        JSONObjectFactory _objectFactory;
+    };
+
+    TEST(Core_Socket, StreamJSON)
+    {
+        IPTestAdministrator::OtherSideMain otherSide = [](IPTestAdministrator & testAdmin) {
+            Core::SocketServerType<JSONConnector<Core::JSON::IElement>> jsonSocketServer(Core::NodeId(StreamJsonTest::g_connector));
+            jsonSocketServer.Open(Core::infinite);
+            testAdmin.Sync("setup server");
+            sleep(2);
+            while(!StreamJsonTest::g_done);
```

Comment:
I guess due to the name temporary fix the idea is to get rid of the active wait loop and state sharing of this between tests. As an alternative you could look at std::condition_variable